src/game.cpp

```diff
@@ -1791,8 +1791,110 @@ void game::update_weather()
+                heat_intensity = ffire;
+            } else if( g->m.tr_at( dest ).loadid == tr_lava ) {
+                heat_intensity = 3;
+            }
+            if( heat_intensity == 0 || !g->m.sees( location, dest, -1 ) ) {
+                // No heat source here
+                continue;
+            }
+            // Ensure fire_dist >= 1 to avoid divide-by-zero errors.
+            const int fire_dist = std::max( 1, square_dist( dest, location ) );
+            fires.emplace_back( std::make_pair( heat_intensity, fire_dist ) );
+            if( fire_dist <= 1 ) {
+                // Extend limbs/lean over a single adjacent fire to warm up
+                best_fire = std::max( best_fire, heat_intensity );
+            }
+        }
+
+        for( const auto &intensity_dist : fires ) {
+            const int intensity = intensity_dist.first;
+            const int distance = intensity_dist.second;
+            temp_mod = 6 * intensity * intensity / distance;
+        }
+
+        // Heat from hot air generated by fires/lava
+        const trap &trap_at_pos = g->m.tr_at( location );
+        // directly on fire/lava tiles
+        int tile_strength = g->m.get_field_strength( location, fd_fire );
+        if( tile_strength > 0 || trap_at_pos.loadid == tr_lava ) {
+            temp_mod += 300;
+        }
+        // hot air of a fire/lava
+        tile_strength = g->m.get_field_strength( location, fd_hot_air1 );
+        switch( tile_strength ) {
+            case 3:
+                temp_mod += 10;
+                break;
+            case 2:
+                temp_mod += 6;
+                break;
+            case 1:
+                temp_mod += 2;
+                break;
+            default:
+                break;
+        }
+        tile_strength = g->m.get_field_strength( location, fd_hot_air2 );
+        switch( tile_strength ) {
+            case 3:
+                temp_mod += 20;
+                break;
+            case 2:
+                temp_mod += 16;
+                break;
+            case 1:
+                temp_mod += 6;
+                break;
+            default:
+                break;
+        }
+        tile_strength = g->m.get_field_strength( location, fd_hot_air3 );
+        switch( tile_strength ) {
+            case 3:
+                temp_mod += 70;
+                break;
+            case 2:
+                temp_mod += 40;
+                break;
+            case 1:
+                temp_mod += 16;
+                break;
+            default:
+                break;
+        }
+        tile_strength = g->m.get_field_strength( location, fd_hot_air4 );
+        switch( tile_strength ) {
+            case 3:
+                temp_mod += 160;
+                break;
+            case 2:
+                temp_mod += 100;
+                break;
+            case 1:
+                temp_mod += 70;
+                break;
+            default:
+                break;
+        }
```

Comment:
Since the structure is very similar over each of the switch blocks, can make use of a lambda:
```
auto tile_strength_mod = []( const tripoint &loc, field_id fld, int case_1, int case_2, int case_3 ){
    int strength = g->m.get_field_strength( loc, fld );
    int cases[3] = { case_1, case_2, case_3 };
    return ( strength > 0 && strength < 4 ) ? cases[ strength - 1 ] : 0;
};
```
Using the lambda:
```
temp_mod += tile_strength_mod( location, fd_hot_air1,  2,   6,  10 );
temp_mod += tile_strength_mod( location, fd_hot_air2,  6,  16,  20 );
temp_mod += tile_strength_mod( location, fd_hot_air3, 16,  40,  70 );
temp_mod += tile_strength_mod( location, fd_hot_air4, 70, 100, 160 );
```

Reply:
Those are code shenanigans I never suspected i'd be using with my C++ skill level, so I'm thankful for the idea, and of course it is a lot cleaner.

Reply:
Actually, looking at how field strength is processed don't even need the conditional. Strength will always be in the range [1..3], so both conditions will always be true.
Can change the return line to `return cases[ strength - 1 ];`